EIPS/eip-4361.md

```diff
@@ -0,0 +1,304 @@
+2. The signature is then presented to the server, which checks the signature's validity and message content.
+3. Additional fields, including `expiration-time`, `not-before`, `request-id`, `chain-id`, and `resources` may be incorporated as part of authentication for the session.
+4. The server may further fetch data associated with the Ethereum address, such as from the Ethereum blockchain (e.g., ENS, account balances, ERC-20/ERC-721/ERC-1155 asset ownership), or other data sources that may or may not be permissioned.
+
+
+### Example message to be signed
+```
+service.org wants you to sign in with your Ethereum account:
+0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
+
+I accept the ServiceOrg Terms of Service: https://service.org/tos
+
+URI: https://service.org/login
+Version: 1
+Nonce: 32891756
+Issued At: 2021-09-30T16:25:24Z
+Chain ID: 1
+Resources:
+- ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/
+- https://example.com/my-web2-claim.json
+```
+
+### Informal Message Template
+A Bash-like informal template of the full message is presented below for readability and ease of understanding. Field descriptions are provided in the following section. A full ABNF description is provided in the section thereafter.
+```
+${domain} wants you to sign in with your Ethereum account:
+${address}
+
+${statement}
+
+URI: ${uri}
+Version: ${version}
+Nonce: ${nonce}
+Issued At: ${issued-at}
+Expiration Time: ${expiration-time}
+Not Before: ${not-before}
+Request ID: ${request-id}
+Chain ID: ${chain-id}
+Resources:
+- ${resources[0]}
+- ${resources[1]}
+...
+- ${resources[n]}
+```
+### Message Field Descriptions
+- `domain` is the [RFC 4501](https://datatracker.ietf.org/doc/html/rfc4501) `dnsauthority` that is requesting the signing.
+- `address` is the Ethereum address performing the signing conformant to capitalization encoded checksum specified in [EIP-55](https://eips.ethereum.org/EIPS/eip-55) where applicable.
+- `statement` (optional) is a human-readable ASCII assertion that the user will sign, and it must not contain `'\n'`.
+- `uri` is an [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986) URI referring to the resource that is the subject of the signing (as in the _subject of a claim_).
+- `version` is the current version of the `message`, which MUST be `1` for this specification.
+- `nonce` is a randomized token used to prevent replay attacks, at least 8 alphanumeric characters.
+- `issued-at` is the [ISO 8601](https://datatracker.ietf.org/doc/html/rfc3339) datetime string of the current time.
+- `expiration-time` (optional) is the [ISO 8601](https://datatracker.ietf.org/doc/html/rfc3339) datetime string that, if present, indicates when the signed authentication message is no longer valid.
```

Comment:
Maybe rename to `not-after` for symmetry with `not-before`?

---

Does if the time is exactly equal to `expiration-time`, is the signature valid?

Reply:
Great naming suggestion!

Also, in regards to is the "signature valid" I think that's up to the implementer of the authentication service, no? Can't really be enforced in the message schema.

Reply:
Thanks for your suggestion, I like it a lot semantically. However, I think I'd prefer to stick to `expiration-time` to keep similarity to JWTs (https://datatracker.ietf.org/doc/html/rfc7519) which would make it more ergonomic to developers already familiar with the terminology there. A lot of the spec was inspired by fields set as precedence in JWTs. Does that make sense?

Reply:
Interpreting your emoji as acceptance, feel free to reopen if you'd like to talk about this more. Thanks!